backend/benefit/applications/services/ahjo_integration.py

```diff
@@ -71,119 +59,96 @@ def _get_template(path):
 
 def prepare_pdf_files(apps):
     pdf_files = []
-    # SINGLE FILE PER APP
-    for app in apps:
-        pdf_files.append(generate_single_file(app))
+    # SINGLE COMPANY/ASSOCIATION PER DECISION PER FILE
+    accepted_apps = [app for app in apps if app.status == ApplicationStatus.ACCEPTED]
+    rejected_apps = [app for app in apps if app.status == ApplicationStatus.REJECTED]
+    accepted_groups = {}
+    for app in accepted_apps:
+        accepted_groups.setdefault(app.company, []).append(app)
+    for group, grouped_accepted_apps in accepted_groups.items():
+        pdf_files.append(generate_single_approved_file(group, grouped_accepted_apps))
+
+    declined_groups = {}
+    for app in rejected_apps:
+        declined_groups.setdefault(app.company, []).append(app)
+    for group, grouped_rejected_apps in declined_groups.items():
+        pdf_files.append(generate_single_declined_file(group, grouped_rejected_apps))
 
     # COMPOSED FILES
-    pdf_files += generate_composed_files(apps)
+    pdf_files += generate_composed_files(accepted_apps, rejected_apps)
 
     return pdf_files
 
 
-def generate_single_file(app):
-    # Association
+def generate_single_declined_file(company, apps):
+    template_config = JINJA_TEMPLATES_SINGLE[TEMPLATE_ID_BENEFIT_DECLINED]
+    file_name = template_config["file_name"].format(company_name=company.name)
+    temp = _get_template(template_config["path"])
+    html = temp.render(apps=apps)
+    single_pdf = pdfkit.from_string(html, False)
+    return file_name, single_pdf, html
+
+
+def generate_single_approved_file(company, apps):
+    # FIXME: Need to change the logic later when we have multiple benefit per application
+    # Association without business activity
```

Comment:
I asked the PO's in Kanslia teams chat to provide an example where this scenario occurs